compaction/compaction_manager.cc

```diff
@@ -938,6 +940,43 @@ bool compaction_manager::can_perform_regular_compaction(replica::table* t) {
     return true;
 }
 
+future<> compaction_manager::maybe_wait_for_sstable_count_reduction(replica::table *t) {
+    if (!can_perform_regular_compaction(t)) {
+        co_return;
```

Comment:
I am thinking that major perhaps shouldn't completely block regular compactions. it should take write lock, to wait for ongoing regular, but as soon as it takes write lock, then perhaps should switch to read lock, to allow compaction on newly created files, to not affect read amplificaiton on L0.

Reply:
It's terrible really that major blocks compaction on newly created files. I think that's a very important thing to fix. Will open an issue.

Reply:
issue: https://github.com/scylladb/scylla/issues/10961